File: src/infrastructure/storage.shared/IOutbox.cs
#nullable enable
using System.Collections.Generic;
using System.Data;
using System.Threading.Tasks;
using core.fs;
using core.Shared;
using Microsoft.FSharp.Core;

namespace storage.shared
{
    public interface IOutbox
    {
        Task<FSharpResult<Unit,ServiceError>> AddEvents(List<AggregateEvent> e, IDbTransaction? tx);
    }
}

File: src/infrastructure/storage.shared.fs/IOutbox.fs
namespace storage.shared

open System.Collections.Generic
open System.Data
open System.Threading.Tasks
open core.fs
open core.Shared
open Microsoft.FSharp.Core

type IOutbox =
    abstract member AddEvents : e:List<AggregateEvent> * tx:IDbTransaction -> Task<Result<unit, ServiceError>>
